playbooks/robusta_playbooks/krr.py

```diff
@@ -296,57 +300,207 @@ def _generate_prometheus_secrets(prom_config: PrometheusConfig) -> List[KRRSecre
+                namespace=scan.object.namespace,
+                name=scan.object.name,
+                kind=scan.object.kind,
+                container=scan.object.container,
+                content=[
+                    {
+                        "resource": resource,
+                        "allocated": {
+                            "request": scan.object.allocations["requests"][resource],
+                            "limit": scan.object.allocations["limits"][resource],
+                        },
+                        "recommended": {
+                            "request": scan.recommended.requests[resource].value,
+                            "limit": scan.recommended.limits[resource].value,
+                        },
+                        "priority": {
+                            "request": scan.recommended.requests[resource].priority,
+                            "limit": scan.recommended.limits[resource].priority,
+                        },
+                        "info": scan.recommended.info.get(resource),
+                        "metric": scan.metrics.get(resource).dict() if scan.metrics.get(resource) else {},
+                        "description": krr_scan.description,
+                        "strategy": krr_scan.strategy.dict() if krr_scan.strategy else None,
+                        "warnings": scan.object.warnings,
+                        "current_pod_count": scan.object.current_pod_count,
+                    }
+                    for resource in krr_scan.resources
+                ],
+            )
+            for scan in krr_scan.scans
+        ],
+        config=config_json,
+    )
+
+@action
+def fail_krr_scan(event: JobEvent, params: FailedScanParams):
+    job = event.get_job()
+    if not job:
+        logging.error(f"cannot run fail_krr_scan on alert with no job object: {event}")
+        return
+    logging.debug("failed scan triggered %s", job.metadata.name)
+    scan_id = params.scan_id
+    if not scan_id:
+        scan_id = job.metadata.annotations.get("scan-id")
+
+    start_time = params.start_time
+    if not start_time:
+        start_time = job.metadata.annotations.get("start-time")
+
+    if not start_time or not scan_id:
+        logging.error(f"cannot run fail_krr_scan, scan_id or start_time are missing: {event}")
+        return
+
+    metadata = {"job": {"name": f"krr-job-{params.scan_id}", "namespace": INSTALLATION_NAMESPACE}}
+    _emit_failed_scan_event(event, scan_id, start_time, metadata, "Krr Job Failed", None, None)
+
+@action
+def process_scan(event: ExecutionBaseEvent, params: ProcessScanParams):
+    if params.scan_type.lower() != "krr":
+        logging.warning(f"Processing scans not supported for type: {params.scan_type}")
+        return
+
+    logging.info(f"Received process_scan request")
+    logging.debug(f"process scan contents {params}")
+
+    metadata = {"job": {"name": f"krr-job-{params.scan_id}", "namespace": INSTALLATION_NAMESPACE}}
+
+    try:
```

Comment:
the code from here until the end looks very similar krr_scan action
Can it be reused?